### PR TITLE
Add array validator

### LINKED ISSUE #1: add validator for arrays
{open for discussion}

``` ruby
  validates :user_ids, array: true
  validates :user_ids, array: { in: [10, 11, 12, 13] }
  validates :user_ids, array: { in: :fetch_valid_ids_from_db }
  # ....
  private

  def fetch_valid_ids_from_db
    Users.active.pluck(:id)
  end
```

### PR DESCRIPTION
**Pull request by kgrz**
Fixes #1

### CODE REVIEW

**Comment by kgrz** [comment]
(Forked off of the pending PR)

**Comment by iffyuva** [comment]
okie, merge latest master onto this one again!

File: README.md [hunk 1]
```diff
@@ -125,6 +125,56 @@ class EventForm
...
+
+    validates [1, 2], verify_array: { not_in: [4, 5] }      # => true
+    validates [1, 2], verify_array: { not_in: [1, 5] }      # => true
+
+    # :eq is a strict validator which will result true only if the given
+    # array contains the exact same elements (irrespective of order)
+    # compared to the verification array.
+
+    validates [1, 2], verify_array: { eq: [2, 1] }          # => true
+    validates [1, 2], verify_array: { eq: [3, 1] }          # => false
```
Comment (iffyuva) [inline README.md:160]: is this one same as:

``` ruby
  validates [1, 2], verify_array: { in: [3, 1] }
```

File: README.md [hunk 2]
```diff
@@ -125,6 +125,56 @@ class EventForm
...
+    end
+
+    # or, specify a Proc that returns an array
+    class EventInvitees < ActiveRecord::Base
+      validates :invited_users, verify_array: { in: ->{ [1, 25, 155] } }
+    end
+
+
+The only caveat here is that the verification item (symbol/proc) should
+return an Array.
```
Comment (iffyuva) [inline README.md:185]: in this case, should we have a check in validator verifying the return value of proc?

File: README.md [hunk 3]
```diff
@@ -125,6 +125,56 @@ class EventForm
...
 ```
 
+### Array Validator
+
+It is quite common to check whether an array of attributes is a subset
+of another array -- a list of valid IDs, for example. The `VerifyArray`
+validator handles the three cases of `:in`, :not_in` and `:eq`. The
+abstract functionality is as follows:
+
+    validates [1, 2, 3], verify_array: { in: [1, 2, 3, 4] } # => true
```
Comment (iffyuva) [inline README.md:142]: there is no

``` ruby
  validates :ids, verify_array: true
```

does ^ work?

File: lib/rails_legit/verify_array_validator.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      end
+    end
+
+    private
+
+    def compare_both_arrays(record_array, verification_array, attribute, type_of_comparision, record)
+      record_array_set = SortedSet.new(record_array)
+      verification_array_set = SortedSet.new(verification_array)
+
+      if type_of_comparision == :in
```
Comment (iffyuva) [inline lib/rails_legit/verify_array_validator.rb]: i feel there shouldn't be a series of if-elses. is it possible to have combination of comparisons? say, if i want to do something like this:

``` ruby
  validates :ids, verify_array: { in: :user_ids, not_in: :suspended_user_ids }
  #...
  def user_ids
    User.all.map(&:id).map(&:to_s)
  end

  def suspended_user_ids
    # assume suspended is a scope.
    User.suspended.map(&:id).map(&:to_s)
  end
```

File: README.md (hunk 1 above)
Comment (kgrz) [inline README.md:160]: Yes.

File: README.md (hunk 2 above)
Comment (kgrz) [inline README.md:185]: We do. We raise an error if the value of proc is not an Array. Although I'm not sure if we should be raising an error or adding the error to the record.

File: README.md (hunk 3 above)
Comment (kgrz) [inline README.md:142]: Yes, we have a basic check

File: lib/rails_legit/verify_array_validator.rb (hunk 4 above)
Comment (kgrz) [inline lib/rails_legit/verify_array_validator.rb]: Hmm, will add some test cases.

File: README.md (hunk 3 above)
Comment (iffyuva) [inline README.md:142]: i guess, we should update readme in that case.

Comment (kgrz) [inline README.md:142]: will do that.

**Comment by iffyuva** [comment]
i tried this one out in one of existing projects. it works, but has one caveat: if i pass a method as a symbol, and if the method is private, validator fails to work.

``` ruby
  validates :ids, verify_array: {in: :user_ids}
  #...

  private
  def user_ids
    User.all.map(&:id).map(&:to_s)
  end
```

can you check?

**Pull request merged**